Updated input handling to fix ImGui.Net breaking changes.

NoahGWood opened the pull request:
Fixed errors caused by changes in latest [ImGuiNet](https://github.com/ImGuiNET/ImGui.NET/issues/451) * enabled support for docking.

NoahGWood committed:
Updated input handling to fix ImGui.Net breaking changes.
Enabled support for docking.

NoahGWood committed:
Combined KeyTranslator.cs with ImGuiController.cs to ensure useable standalone per [PR 29 Review](https://github.com/NogginBops/ImGui.NET_OpenTK_Sample/pull/29#pullrequestreview-1744615513)

Amending:
Updated input handling to fix ImGui.Net breaking changes.
Enabled support for docking.

NogginBops reviewed (changes requested):
Thanks a lot for the PR!

A goal with this repo is to make `ImGuiController.cs` useable standalone by just copying the file, this means that the `KeyTranslator.cs` should be part of the `UpdateImGuiInput()` method (it's fine if the method becomes really long).

NoahGWood committed:
Merge branch 'opentk4.0' of https://github.com/NoahGWood/ImGui.NET_OpenTK_Sample into opentk4.0

NoahGWood committed:
Delete Dear ImGui Sample/KeyTranslator.cs

I've been using github since 2017 and I still don't know why it takes me 40+ commits and PR's to change one file, but I'm blaming Visual Studio this time

NoahGWood commented:
Should be fixed :)

zaafar commented on Dear ImGui Sample/ImGuiController.cs:
```diff
@@ -588,4 +564,132 @@ public static void CheckGLError(string title)
+    {
+        public static ImGuiKey Translate(Keys key)
+        {
+            switch (key)
```
[optional] you can reduce the length a bit if you do something like this.
https://github.com/zaafar/ClickableTransparentOverlay/blob/master/ClickableTransparentOverlay/ImGuiInputHandler.cs#L161:L172

NogginBops committed:
Simplified key translation code
